### PR TITLE
feat: emits "error" event when the confirmation fails

### PR DESCRIPTION
**Pull request by vrolland**

### CODE REVIEW

File: packages/request-client.js/src/api/request.ts [hunk 1]
```diff
@@ -26,6 +26,8 @@ export default class Request {
   private contentDataExtension: ContentDataExtension | null;
   private emitter: EventEmitter;
 
+  private confirmationErrorOccurred: boolean = false;
```
Comment (ltacker) [inline packages/request-client.js/src/api/request.ts]: Could be explained here why this variable is used

File: packages/request-client.js/src/api/request.ts [hunk 2]
```diff
@@ -133,9 +143,13 @@ export default class Request {
...
 
-    acceptResult.on('confirmed', async () => {
-      requestData.emit('confirmed', await this.refresh());
-    });
+    acceptResult
+      .on('confirmed', async () => {
+        requestData.emit('confirmed', await this.refresh());
+      })
+      .on('error', error => {
+        this.emitter.emit('error', error);
```
Comment (ltacker) [inline packages/request-client.js/src/api/request.ts]: Mhh why `confirmationErrorOccurred` is not set here ?

File: packages/request-client.js/src/api/request.ts [hunk 3]
```diff
@@ -171,9 +185,13 @@ export default class Request {
...
 
-    cancelResult.on('confirmed', async () => {
-      requestData.emit('confirmed', await this.refresh());
-    });
+    cancelResult
+      .on('confirmed', async () => {
+        requestData.emit('confirmed', await this.refresh());
+      })
+      .on('error', error => {
+        this.emitter.emit('error', error);
```
Comment (ltacker) [inline packages/request-client.js/src/api/request.ts]: Same as above comment (and same comment for all these kind of line), why not setting `confirmationErrorOccurred` ?

File: packages/request-client.js/src/api/request.ts [hunk 4]
```diff
@@ -574,6 +624,10 @@ export default class Request {
    * @returns The updated request data
    */
   public getData(): Types.IRequestDataWithEvents {
+    if (this.confirmationErrorOccurred) {
+      throw Error('request confirmation failed');
```
Comment (ltacker) [inline packages/request-client.js/src/api/request.ts]: The message should be more explicit
We could define `confirmationErrorOccurred` as a string and insert the error occurred inside to print it here

File: packages/request-client.js/src/api/request.ts [hunk 5]
```diff
@@ -603,6 +657,9 @@ export default class Request {
   public async refresh(
     requestAndMeta?: RequestLogicTypes.IReturnGetRequestFromId,
   ): Promise<Types.IRequestDataWithEvents> {
+    if (this.confirmationErrorOccurred) {
+      throw Error('request confirmation failed');
```
Comment (ltacker) [inline packages/request-client.js/src/api/request.ts]: Same as above

File: packages/request-client.js/src/api/request.ts (hunk 2 above)
Comment (vrolland) [reply packages/request-client.js/src/api/request.ts]: It is only for the creation, I renamed the variable

File: packages/request-client.js/src/api/request.ts (hunk 3 above)
Comment (vrolland) [reply packages/request-client.js/src/api/request.ts]: same same

File: packages/request-client.js/src/api/request.ts (hunk 1 above)
Comment (vrolland) [reply packages/request-client.js/src/api/request.ts]: documented and renamed

File: packages/request-client.js/src/api/request.ts (hunk 4 above)
Comment (vrolland) [reply packages/request-client.js/src/api/request.ts]: We get the error from an event already. I think it's enough.

**Pull request merged**